File: 2020/Python/d05.py
data = open("../input/05.txt", 'r').read()
long = [int(x) for x in range(128)]
short = [int(x) for x in range(8)]

# Parse binary number
def parse(ticket, matrix, n, low):
    s = n
    for c in ticket:
        s //= 2
        if c == low: matrix = matrix[:s]
        else: matrix = matrix[s:]
    return matrix[0]

# Part 1
_ids = list()
for ticket in data.split('\n'):
    column = parse(ticket[:7], long, len(long), 'F')
    row = parse(ticket[7:], short, len(short), 'L')
    _ids.append(column * 8 + row)

print(f"Part 1: {max(_ids)}")

# Part 2
_ids = sorted(_ids)
for i in range(len(_ids)):
    if i + _ids[0] != _ids[i]:
        print(f"Part 2: {_ids[i]-1}")
        break

File: 2020/C#/d05.cs
using System;
using System.Collections.Generic;
using System.Linq;

namespace _2020
{
    class d05
    {
        public static void solve(){
            Console.WriteLine("Day 05");
            string data = System.IO.File.ReadAllText(@"../input/05.txt");
            var seats = new List<double>();
            // Part 1
            foreach (string line in data.Split('\n')){
                string nLine = line.Replace('F', '0')
                    .Replace('B', '1')
                    .Replace('L', '0')
                    .Replace('R', '1');
                
                double colum = Convert.ToInt32(nLine[0..7], 2);
                double row = Convert.ToInt32(nLine[7..10], 2);
                seats.Add((colum * 8) + row);
            }
            Console.WriteLine($"Part 1: {seats.Max()}");
            // Part 2
            seats.Sort();
            for (int i=0; i<seats.Count; i++){
                if ((i + 45) != seats[i]){                   
                    Console.WriteLine($"Part 2: {(seats[i]-1)}");
                    break;
                    
                }
            }
        }
    }
}